AK/Platform.h

```diff
@@ -182,6 +182,11 @@
 #endif
 #define RETURNS_NONNULL __attribute__((returns_nonnull))
 
+#ifdef NO_SANITIZE_COVERAGE
+#    undef NO_SANITIZE_COVERAGE
+#endif
+#define NO_SANITIZE_COVERAGE __attribute__((no_sanitize_coverage))
```

Comment:
Going off the Azure pipelines, it looks like clang wants 
`__attribute__((no_sanitize("coverage")))`

https://releases.llvm.org/18.1.0/tools/clang/docs/SanitizerCoverage.html#disabling-instrumentation-with-attribute-no-sanitize-coverage

https://dev.azure.com/SerenityOS/SerenityOS/_build/results?buildId=46221&view=logs&j=1427867c-05b9-544b-63a5-81f0b096b4ff&t=99e26231-27cd-5190-be33-4ff6ca9ae751&l=400

Reply:
well that's annoying because gcc specifically refuses that x)

```
/home/space/repos/serenity/./Kernel/Tasks/Process.h:229:54: error: 'coverage' attribute directive ignored [-Werror=attributes]
  229 |     NO_SANITIZE_COVERAGE KCOVInstance* kcov_instance()
```

Though [this should work, according to the docs](https://gcc.gnu.org/onlinedocs/gcc-12.2.0/gcc/Common-Function-Attributes.html#index-no_005fsanitize-function-attribute).. urgs gcc.

Reply:
I went for the second, let's see how this goes.